astropy/table/table.py

```diff
@@ -593,7 +620,7 @@ class Table:
         Copy the input data. If the input is a Table the ``meta`` is always
         copied regardless of the ``copy`` parameter.
         Default is True.
-    rows : numpy ndarray, list of list, optional
+    rows : numpy ndarray, list of list, sequence of ``Row``, optional
```

Comment:
Addition of new accepted type as input also feel like a feature, so I didn't mark this as backport but Table maintainers can always change that if they want.

Thanks!

Reply:
Initializing from a sequence of Row was always intended to work, so this is just documenting that correctly (and fixing a bug).

Reply:
So, @taldcroft , should we backport?

Reply:
Yes on the backport in concept, though to be honest I don't know where it will be backported. I thought there is only 6.0.1 which is the next bugfix release forward from 6.0.0.

Also, of course, I haven't reviewed the code yet!